### ISSUE TITLE
Observable.Return() concat with Interval, the latter returns nothing

### ISSUE THREAD

**Issue by luxl** [Cysharp/R3#217]
```c#
var r1 = Observable.Return(1);
var r2 = Observable.Interval(TimeSpan.FromSeconds(1)).Index();

r1.Concat(r2).Subscribe(Console.WriteLine);

await Task.Delay(TimeSpan.FromDays(1)); // wait
```
the above code only print "1", did not print anything in r2, is this expected? or did I miss anything?
the above example works in Rx.NET, not in R3.
But if I concat two Intervals, it worked as expected, like code below:

```c#
var r1 = Observable.Interval(TimeSpan.FromMilliseconds(500)).Take(1).Index();
var r2 = Observable.Interval(TimeSpan.FromSeconds(1)).Index();

r1.Concat(r2).Subscribe(Console.WriteLine);  

await Task.Delay(TimeSpan.FromDays(1)); // wait
```
It will first print all in r1 and then print all in r2

**Comment by kamend** [comment]
This might be the same issue as here: https://github.com/Cysharp/R3/issues/232

**Comment by neuecc** [comment]
Thank you for the Issue report and sorry for the delay in responding.
The fix has been completed in v1.1.15.
